### PR TITLE
Refactor: improve compatibility with opa clients and tests

### PR DESCRIPTION
**Pull request by MatteoGranziera**
- Improved compatibility with opa clients 
- refactored tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,62 +9,53 @@ __Made with ❤️ at&nbsp;&nbsp;[<img src="https://assets.treedom.net/image/upl
...
-})
+const { url } = await startStandaloneServer(server, {
+  // require an http.IncomingMessage implementation 
+  context: async ({ req }) => ({ request: req }),
+});
 ```
 
 ## OPA policy input
 
 This plugin queries OPA providing the following properties as `input`
```
Comment (OancaAndrei) [inline README.md]:
```suggestion
This plugin queries OPA providing the following properties as `input`:
```

File: README.md [hunk 2]
```diff
@@ -89,6 +80,86 @@ allow if {
 }
 ```
 
+## Headers forwarding
+
+If you need to forward the headers to OPA you can use the `requestContextField` option to specify the name of the request context field
```
Comment (OancaAndrei) [inline README.md]:
```suggestion
If you need to forward the headers to OPA you can use the `requestContextField` option to specify the name of the request context field.
```

File: README.md [hunk 3]
```diff
@@ -89,6 +80,86 @@ allow if {
...
+
+To do that you need to manually build the context by adding the request object. 
+
+> Currently the request should have a `headers` property with `http.IncomingHttpHeaders` type. 
+> Compatible with `FastifyRequest<...>` and `http.IncomingRequest`
```
Comment (OancaAndrei) [inline README.md]:
```suggestion
> Compatible with `FastifyRequest<...>` and `http.IncomingRequest`.
```

File: README.md [hunk 4]
```diff
@@ -89,6 +80,86 @@ allow if {
...
+      }
+    },
+  })
+
+```
+
+
 ## Custom directive
 
 The authorization directive can be customized registering a custom one in the schema and specifying its name in the plugin configuration
```
Comment (OancaAndrei) [inline README.md]:
```suggestion
The authorization directive can be customized registering a custom one in the schema and specifying its name in the plugin configuration:
```

File: README.md [hunk 5]
```diff
@@ -9,62 +9,53 @@ __Made with ❤️ at&nbsp;&nbsp;[<img src="https://assets.treedom.net/image/upl
...
 ```
 
 ## OPA policy input
 
 This plugin queries OPA providing the following properties as `input`
 
-- `headers` the Fastify headers object
-- `parent` the Mercurius parent object of the field/object which got queried
-- `args` the Mercurius args object of the field/object which got queried
+- `headers` the headers object, this require a context request forwarding
```
Comment (OancaAndrei) [inline README.md]:
```suggestion
- `headers` the headers object, this requires a context request forwarding
```

**Review by OancaAndrei — APPROVED** [review]

**Review by OancaAndrei — APPROVED** [review]

**Review by OancaAndrei — APPROVED** [review]

File: .prettierrc [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+    "semi": false,
+    "singleQuote": true,
+    "trailingComma": "es5",
+    "import/order": [
```
Comment (vivedo) [inline .prettierrc:5]: this looks an eslint plugin config, is it supposed to be in prettier config file?

**Review by vivedo — APPROVED** [review]
lgtm

**Pull request merged**